File: packages/frontend/types/playlist.ts
export enum PlaylistContext {
  WORK = "work",
  BREAK = "break",
}

export interface AudioTrack {
  id: number;
  playlist_id: number;
  title: string;
  file_path: string; // URL or path
  duration?: number;
  position: number;
  created_at: string;
}

export interface AudioTrackCreate {
  title: string;
  file_path: string;
  duration?: number;
}

export interface TrackReorder {
  track_id: number;
  new_position: number;
}

export interface Playlist {
  id: number;
  name: string;
  context: PlaylistContext;
  tracks: AudioTrack[];
  created_at: string;
  updated_at: string;
}

File: packages/backend/app/schemas/playlist.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field

from app.models.playlist import PlaylistContext


class AudioTrackBase(BaseModel):
    title: str = Field(..., min_length=1)
    file_path: str = Field(..., min_length=1)
    duration: Optional[int] = None


class AudioTrackCreate(AudioTrackBase):
    pass


class AudioTrackResponse(AudioTrackBase):
    id: int
    playlist_id: int
    position: int
    created_at: datetime

    class Config:
        from_attributes = True


class PlaylistBase(BaseModel):
    name: str = Field(..., min_length=1)
    context: PlaylistContext


class PlaylistCreate(PlaylistBase):
    pass


class PlaylistResponse(PlaylistBase):
    id: int
    tracks: List[AudioTrackResponse] = []
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class TrackReorder(BaseModel):
    track_id: int
    new_position: int

